GVFS/GVFS.Common/InstallerPreRunChecker.cs

```diff
@@ -0,0 +1,243 @@
+using GVFS.Common;
+using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+
+namespace GVFS.Upgrader
+{
+    public class InstallerPreRunChecker
+    {
+        private static readonly string[] BlockingProcessList = { "GVFS", "GVFS.Mount", "git", "ssh-agent", "bash", "gitk", "git-bash" };
+
+        private ITracer tracer;
+
+        public InstallerPreRunChecker(ITracer tracer)
+        {
+            this.tracer = tracer;
+        }
+
+        public bool TryRunPreUpgradeChecks(GitVersion gitVersion, out string error)
+        {
+            this.tracer.RelatedInfo("Checking if GVFS upgrade can be run on this machine.");
+
+            if (this.IsUnattended())
+            {
+                error = "`gvfs upgrade [--confirm]` is not supported in unattended mode";
+                this.tracer.RelatedError($"{nameof(TryRunPreUpgradeChecks)}: {error}");
+                return false;
+            }
+
+            if (this.IsDevelopmentVersion())
+            {
+                error = "Cannot run upgrade when development version of GVFS is installed.";

```

Comment:
>>Why not? What's the harm in it?
I can remove this, but the initial thought was Developers would be building from the latest sources and running the latest version even though their VFSForGit version numbers begin with "0".

>>I know this is not a developer feature but as a developer I would want to run to install any version with a gvfs install --version <some identifier> or gvfs install --debug because I find myself jumping around versions quit a bit, but allowing me to at least run upgrade to get the latest would be a start.

I will make a note. We can review and consider adding this option in a future PR.